internal/pkg/manifest/svc.go

```diff
@@ -442,73 +456,91 @@ func (s *StringOrFromCFN) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
-func (cfg ImageWithPortAndHealthcheck) exposedPorts(workloadName string) []ExposedPort {
+func (cfg ImageWithPortAndHealthcheck) exposePorts(exposedPorts map[uint16]ExposedPort, workloadName string) {
 	if cfg.Port == nil {
-		return nil
+		return
 	}
-	return []ExposedPort{
-		{
-			Port:                 aws.Uint16Value(cfg.Port),
-			Protocol:             "tcp",
-			ContainerName:        workloadName,
+
+	targetPort := aws.Uint16Value(cfg.Port)
+	if exposedPort, ok := exposedPorts[targetPort]; ok {
```

Comment:
Am I correct that this becomes necessary because we move this `exposePorts` function call to after all the other `exposePorts` (e.g. on nlb, alb...)?

Reply:
Yep!